### PR TITLE
Enhance representation of growth stage in AgContext

### LINKED ISSUE #115: Fix up specification of BBCH growth stage codes
1. Currently called "bbch_code" should be "bbch_growth_stage"
2. Currently specified with gs prefix. This should be made more explicit in the description or with examples, or the gs prefix should be removed.
3. How do we want to allow a range or several BBCH growth stages to be specified? The growth stage is intended to be applied to a single plant, not a plantation.

### PR DESCRIPTION
**Pull request by jnothman**
Fixes #115

### CODE REVIEW

**Comment by jnothman** [comment]
I note that filecmp.cmpfiles in test_deposit is a pain to work with. It nitpicks on subtle changes and can waste a lot of time.

**Comment by jnothman** [comment]
To summarise this PR:
Growth stage is now represented as a range: a pair of integers for min and max BBCH code.
At the moment the process of indexing includes translating the code range to grains descriptive texts, which could be searched on and shown on the datasets page.

In the future I imagine widgets to search by filtering on the min and max growth stage separately (although that might be overkill??)

File: weedcoco/index/indexing.py [hunk 1]
```diff
@@ -67,6 +68,17 @@ def _flatten(src, dst, prefix):
             for field in variable_to_null_fields:
                 if agcontext.get(field) == "variable":
                     del agcontext[field]
+            # textual label for growth stage
+            try:
```
Comment (hlydecker) [inline weedcoco/index/indexing.py:72]: LGTM

**Review by hlydecker — APPROVED** [review]
This looks like a very reasonable way to more clearly/simply represent growth stage, and I can see how this will be easier to interact with going forward.

**Comment by jnothman** [comment]
Thanks for the review @hlydecker

**Comment by hlydecker** [comment]
Sorry for the delay; you two have been very productive and I lost it in a swarm of notifications.

**Pull request merged**